### PR TITLE
[RFR] [SVCS-252] Update provider base path removal to be more strict.

### PR DESCRIPTION
**Pull request by TomBaxter**
## Purpose:

[SVCS-252](https://openscience.atlassian.net/browse/SVCS-252)
Fix poor error handling in Dropbox provider.

## Changes:

update  waterbutler/providers/dropbox/provider.py
  Improve logic for provider base folder removal in def dropbox_conflict_error_handler

## Side effects:

None

[SVCS-252]

### CODE REVIEW

File: waterbutler/providers/dropbox/provider.py [hunk 1]
```diff
@@ -89,7 +89,8 @@ def dropbox_conflict_error_handler(self, data, error_path=''):
         :param str error_path: the path where the error occurred. Base folder will be stripped.
         """
 
-        error_path = '/{}'.format(error_path.lstrip(self.folder))
+        if error_path.startswith(self.folder):
+            error_path = error_path[len(self.folder):]
```
Comment (felliott) [inline waterbutler/providers/dropbox/provider.py]: The old code looks like it prepended a slash to the path, but this new code doesn't.  Is that correct?

Comment (TomBaxter) [inline waterbutler/providers/dropbox/provider.py]: Done.

**Comment by TomBaxter** [comment]
Added test for dropbox_conflict_error_handler.

**Comment by felliott** [comment]
Merged as 11611ea1d2fc06ae288c456a2a2c17fbd321442c

**Pull request closed without merging**